pkg/client/cache/reflector.go

```diff
@@ -337,16 +304,23 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 			return nil
 		}
-		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+
+		if err := r.watchHandler(w, &resourceVersion, stopCh); err != nil {
+			if err != errorStopRequested {
 				glog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedType, err)
 			}
 			return nil
 		}
-		if r.canForceResyncNow() {
-			glog.V(4).Infof("%s: next resync planned for %#v, forcing now", r.name, r.nextResync)
-			return nil
-		}
+		/*
```

Comment:
> If you start watch on say 5, then even though etcd is on version 50, you will get all notifcations betwen 5 and 50.

That's totally not the case, you will receive the next event but there is no playback, and never has been.  I'll let @xiang90 correct my comment with specifics 

The issue rob has been having has been subtle timing issue if we ever break a watch.